clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -113,16 +113,7 @@ private TopicPartitionEntry getPartition(TopicPartition topicPartition) {
         }
 
         private TopicPartitionEntry getOrCreatePartition(TopicPartition topicPartition) {
-            TopicPartitionEntry ent = topicPartitions.get(topicPartition);
-            if (ent == null) {
-                ent = new TopicPartitionEntry();
-                topicPartitions.put(topicPartition, ent);
-            }
-            return ent;
-        }
-
-        private void addPartition(TopicPartition topicPartition) {
-            this.topicPartitions.putIfAbsent(topicPartition, new TopicPartitionEntry());
+            return topicPartitions.putIfAbsent(topicPartition, new TopicPartitionEntry());
```

Comment:
I don't think this refactor is correct. `putIfAbsent` will return **previous** value of the key. That is, if the current partition is null, after `getOrCreatePartition`, it'll return `null`, which is not what we want.

Reply:
Thanks for reminding me of that! I've confused it with `computeIfAbsent`. How about using `computeIfAbsent` as I modified?